src/read.rs

```diff
@@ -68,17 +68,18 @@ impl<R: Read + Debug> Rng for ReadRng<R> {
         impl_uint_from_fill!(u128, 16, self)
     }
     
-    fn try_fill(&mut self, v: &mut [u8]) -> Result<()> {
+    fn try_fill(&mut self, v: &mut [u8]) -> Result<(), Error> {
         if v.len() == 0 { return Ok(()); }
         fill(&mut self.reader, v)
     }
 }
 
-fn fill(r: &mut Read, mut buf: &mut [u8]) -> Result<()> {
+fn fill(r: &mut Read, mut buf: &mut [u8]) -> Result<(), Error> {
     while buf.len() > 0 {
-        match r.read(buf)? {
-            0 => return Err(Error),
-            n => buf = &mut mem::replace(&mut buf, &mut [])[n..],
+        match r.read(buf) {
+            Ok(0) => return Err(Error::new(ErrorKind::Unavailable, None)),
```

Comment:
Nothing you can do with the given error type, but it does feel like the *cause* is missing here

Reply:
It is fixed in the third commit, that gives `ReadRng` better error handling